climakitae/new_core/param_validation/warming_param_validator.py

```diff
@@ -0,0 +1,216 @@
+        warnings.warn(
+            "\n\nInvalid 'warming_levels' parameter. "
+            "\nPlease check the configuration."
+        )
+        return False
+
+    return True
+
+
+def _check_input_types(
+    value: dict[str, Any],
+):
+    if not isinstance(value, dict):
+        warnings.warn(
+            "\n\nWarming Level Processor expects a dictionary of parameters. "
+            "\nPlease check the configuration."
+        )
+        return False
+
+    wl = value.get("warming_levels", UNSET)
+    if (
+        wl is UNSET
+        or not isinstance(wl, list)
+        or not all(isinstance(x, (int, float)) for x in wl)
+    ):
+        warnings.warn(
+            "\n\nInvalid 'warming_levels' parameter. "
+            "\nExpected a list of global warming levels (e.g., [1.5, 2.0])."
+        )
+        return False
+
+    wl_months = value.get("warming_level_months", UNSET)
+    if wl_months is not UNSET:
+        if not isinstance(wl_months, list) or not all(
+            isinstance(x, int) and 1 <= x <= 12 for x in wl_months
+        ):
+            warnings.warn(
+                "\n\nInvalid 'warming_level_months' parameter. "
+                "\nExpected a list of months (1-12). Default: all months."
+            )
+            return False
+
+    wl_window = value.get("warming_level_window", UNSET)
+    if wl_window is not UNSET:
+        if not isinstance(wl_window, int) or wl_window < 0:
+            warnings.warn(
+                "\n\nInvalid 'warming_level_window' parameter. "
+                "\nExpected a non-negative integer (default: 15)."
+            )
+            return False
+
+    return True
+
+
+def _check_query(query: Any) -> bool:
+    """
+    Warming level approach requires we check activity_id and experiment_id
+
+    Activity_id needs to be "WRF" or "LOCA2" or UNSET
+
+    experiment_id needs to be "UNSET".
+
+    """
+    if not isinstance(query, dict):
+        return False
+
+    activity_id = query.get("activity_id", UNSET)
+    experiment_id = query.get("experiment_id", UNSET)
+
+    if activity_id not in ["WRF", "LOCA2", UNSET]:
+        warnings.warn(
+            "\n\nInvalid 'activity_id' parameter. "
+            "\nExpected 'WRF', 'LOCA2', or not passed (UNSET)."
+        )
+        # force the user to fix this. Cannot assume intention here
+        return False
+
+    if experiment_id is not UNSET:
+        warnings.warn(
+            "\n\nWarming level approach requires 'experiment_id' to be UNSET. "
+            "\nModify the query accordingly."
+        )
+        return False
+
+    time_slice = query.get("processes", {}).get("time_slice", UNSET)
+    if time_slice is not UNSET:
+        warnings.warn(
+            "\n\nWarming level approach does not support 'time_slice' in the query. "
+            "\nIt will be ignored."
+        )
+        del query["processes"]["time_slice"]
+
+    return True
+
+
+def _check_wl_values(query: dict[str, Any]) -> bool:
+    """
```

Comment:
Might just be style preference, but some blank lines in the docstring would increase readability.